src/components/ParkingLotForm/index.tsx

```diff
@@ -0,0 +1,410 @@
+import { transformToFormError } from '#utils/errorTransform';
+
+import {
+    requiredCondition,
+    requiredStringCondition,
+    idCondition,
+    urlCondition,
+} from '#utils/validation';
+
+import {
+    enumKeySelector,
+    enumLabelSelector,
+} from '#utils/common';
+
+import {
+    PartialForm,
+    PurgeNull,
+} from '#types';
+
+import {
+    ParkingLotOptionsQuery,
+    ParkingLotQuery,
+    ParkingLotQueryVariables,
+    CreateParkingLotMutation,
+    CreateParkingLotMutationVariables,
+    UpdateParkingLotMutation,
+    UpdateParkingLotMutationVariables,
+} from '#generated/types';
+import styles from './styles.css';
+
+const PARKING_LOT_OPTIONS = gql`
+    query ParkingLotOptions {
+        status: __type(name: "PARKING_LOT_STATUS") {
+            enumValues {
+                name
+                description
+            }
+        }
+    }
+`;
+
+const PARKING_LOT = gql`
+    query ParkingLot($id: ID!) {
+        parkingLot(id: $id) {
+            comments
+            country {
+                id
+                name
+            }
+            id
+            assignedTo {
+                id
+                fullName
+            }
+            status
+            createdBy {
+                id
+                fullName
+            }
+            title
+            url
+        }
+    }
+`;
+
+const CREATE_PARKING_LOT = gql`
+    mutation CreateParkingLot($parkingLot: ParkingLotCreateInputType!) {
+        createParkingLot(data: $parkingLot) {
+            result {
+                id
+            }
+            errors
+        }
+    }
+`;
+
+const UPDATE_PARKING_LOT = gql`
+    mutation UpdateParkingLot($parkingLot: ParkingLotUpdateInputType!) {
+        updateParkingLot(data: $parkingLot) {
+            result {
+                id
+            }
+            errors
+        }
+    }
+`;
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+type WithId<T extends object> = T & { id: string };
+type ParkingLotFormFields = CreateParkingLotMutationVariables['parkingLot'];
+type FormType = PurgeNull<PartialForm<WithId<Omit<ParkingLotFormFields, 'status'> & { status: string }>>>;
+
+type FormSchema = ObjectSchema<FormType>;
+type FormSchemaFields = ReturnType<FormSchema['fields']>;
+
+const schema: FormSchema = {
+    fields: (): FormSchemaFields => ({
+        id: [idCondition],
+        country: [requiredCondition],
+        title: [requiredStringCondition],
+        url: [requiredCondition, urlCondition],
+        assignedTo: [requiredCondition],
+        status: [requiredCondition],
+        comments: [],
+    }),
+};
+
+interface ParkingLotFormProps {
+    className?: string;
+    onParkingLotCreate?: (result: NonNullable<NonNullable<CreateParkingLotMutation['createParkingLot']>['result']>) => void;
+    id?: string;
+    readOnly?: boolean;
+    onParkingLotFormCancel: () => void;
+}
+
+function ParkingLotForm(props: ParkingLotFormProps) {
+    const {
+        onParkingLotCreate,
```

Comment:
The problem name can be simpler.

```suggestion
        onCreate,
```